Readme.md

```diff
@@ -8,11 +8,106 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
+
+    - **Compiler:** Compiler implementation
+
+      - Lexer: Converts source code into `Tokens`
+
+      - Parser: Converts `Tokens` into `ParseNodes`
+
+      - Validator: Compile-time conditions for the parser
+
+      - Importer: Extracts the `TypeSymbols` from the assemblies (using Mono.Cecil) and generates `Metadata` for them
+        > f.e: create all MethodSymbols for a declared `ClassSymbol`
+
+      - Builder: Transforms `ParseNodes` into `Metadata`
+        > f.e: create a Expression tree from a node
+
+      - Generator: Transforms `Symbols` and `Metadata` into output (javascript)
+
+    - **Mscorlib:** Dummy interfaces and classes that emulate the original `mscorlib.dll`. Having the same signatures of the original mscorlib enables cross-compilation.
+
+    - **Scripts:** Javascript implementations of mscorlib. It mostly includes helper methods and class implementations. The compiler binds C# code to JS implementation by using `ScriptAttributes`.
+
+    - **Web:** Web-specific abstractions that can be used when targeting DSharp. This is an extension library and isn't available in normal C#.
+
+    - **Shell:** Shell utility for invoking the compiler.
+
+- ### SDK
+
+    - Crosscompilation for DSharp and NetStandard
+    - Support for modern `csproj` syntax
+
+- ### Tests
+
+    - **Framework:** Implementation of the JS test framework.
+    - **Compiler.Tests:** Tests for the compiler.
+
+## Writing Tests ##
+
+Testing the compiler is quite straightforward: define an input c# file `Code.cs` and the expected result in `Baseline.txt`. The test runner will compare the result with the expected one.
+
+In order to create a new  you need to:
+
+1. Create the test files `Code.cs` and `Baseline.txt` in a relevant subdirectory under the `Source` folder.
+2. Add your test to `CompilerTests.json` pointing to the right files. The categories represent subdirectories inside the `Source` folder.
+
+## Debugging ##
+
+There are two ways of debugging the project: debugging a test or a shell that invokes the compiler.
+
+### Prerequisites ###
+
+- #### Enable all Common Language Runtime Exceptions
+  This can be done under `Debug` -> `Windows` -> `Exception Settings (Ctrl+Alt+E)` and ticking all `Common Language Runtime Exceptions`.
+
+### Debugging a Test ###
+
+Debugging a test is currently non-deterministic (it works depending on the computer, and it may display some artifacts on Visual Studio) but it's worth trying. This is due to xUnit runtime confusing Visual Studio.
+
+In order to debug a test you need to:
+
+1. Rebuild All
+2. In the test explorer, `Right Click` -> `Debug Selected Tests`
+
+### Debugging the Shell ###
+
+Debugging the Shell requires a bit of configuration: setting up the arguments for the invokation.
+
+In order to debug the shell you need to:
+
+1. Rebuild All
+2. In the `Core/DSharp.Shell` project, `Right Click` -> `Properties` -> `Debug` -> `Application arguments` and set the following:
+
+    ```
```

Comment:
Formatting, also can you mark the code block with the syntax `console`

Reply:
formatting is fine, since I want it to be indented. Adding the `console`